os/net/ipv6/uip-nd6.c

```diff
@@ -184,6 +184,13 @@ ns_input(void)
   UIP_STAT(++uip_stat.nd6.recv);
 
 #if UIP_CONF_IPV6_CHECKS
+
+  if (uip_l3_icmp_hdr_len + sizeof(uip_nd6_ns) > uip_len)
+  {
+    LOG_ERR("Not enough space for cast");
```

Comment:
Can you make this log statement here clearer? Maybe "Insufficient data for reading ND6 NS header fields".

Reply:
Addressed this in new commit